lightning/api_clnraw.go

```diff
@@ -767,3 +742,288 @@ func (l *ClnRawLightningAPI) GetInternalChannelsAll(ctx context.Context) (map[st
 
 	return result, nil
 }
+
+// ConnectPeer - API call.
+func (l *ClnRawLightningAPI) ConnectPeer(ctx context.Context, id string) error {
+	var resp ClnConnectResp
+	err := l.connection.Call(ctx, Connect, []string{id}, &resp)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.ID) == 0 {
+		return fmt.Errorf("invalid response")
+	}
+
+	return nil
+}
+
+// GetOnChainAddress - API call.
+func (l *ClnRawLightningAPI) GetOnChainAddress(ctx context.Context) (string, error) {
+	var resp ClnNewAddrResp
+
+	err := l.connection.Call(ctx, NewAddr, []interface{}{}, &resp)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resp.Bech32) == 0 {
+		return "", fmt.Errorf("invalid response")
+	}
+
+	return resp.Bech32, nil
+}
+
+// GetOnChainFunds - API call.
+func (l *ClnRawLightningAPI) GetOnChainFunds(ctx context.Context) (*Funds, error) {
+	const (
+		Confirmed   = "confirmed"
+		Unconfirmed = "uncomfirmed"
+		Mili        = 1000
+	)
+
+	var fundsReply ClnFundsChainResp
+	err := l.connection.Call(ctx, ListFunds, []string{}, &fundsReply)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &Funds{}
+	f.ConfirmedBalance = 0
+	f.TotalBalance = 0
+	f.LockedBalance = 0
+
+	for _, one := range fundsReply.Outputs {
+		amount := int64(ConvertAmount(one.AmountMsat))
+		if !one.Reserved && one.Status == Confirmed {
+			f.ConfirmedBalance += (amount / Mili)
+			f.TotalBalance += (amount / Mili)
+		} else if !one.Reserved && one.Status == Unconfirmed {
+			f.TotalBalance += (amount / Mili)
+		} else if one.Reserved && (one.Status == Unconfirmed || one.Status == Confirmed) {
+			f.LockedBalance += (amount / Mili)
+		}
+	}
+
+	return f, nil
+}
+
+// SendToOnChainAddress - API call.
+func (l *ClnRawLightningAPI) SendToOnChainAddress(ctx context.Context, address string, sats int64, useUnconfirmed bool, urgency Urgency) (string, error) {
+	var reply ClnWithdrawResp
+
+	feerate := "normal"
+	switch urgency {
+	case Urgent:
+		feerate = "urgent"
+	case Normal:
+		feerate = "normal"
+	case Low:
+		feerate = "slow" // slow not low, beware!
```

Comment:
The comment is just because "low" and "slow" are so common and somebody might think the feerate string is a typo for "low". CLN allows you to pass it "slow" string as parameter for feerate.

And we have an abstraction {Urgent, Normal, Low} "urgency". Possibly this naming needs to be improved.